app.py

```diff
@@ -151,100 +169,77 @@ async def delete_user(user_id):
-                              (user[0], refresh_token))
-            await cur.close()
-            pool.close()
-    return jsonify({'access_token': access_token, 'refresh_token': refresh_token, 'access_expire': access_expire,
-                    'refresh_expire': refresh_expire}), 200
+    user_dict = dict(zip(("user_id", "user_name", "email", "password"), user))
+    if not user or not check_password_hash(user_dict["password"], password):
+        return jsonify({"error": "Invalid username or password."}), 401
+    return create_and_return_token(user_dict["user_id"])
 
 
 # Obtain new access token from refresh token endpoint
-@app.route('/refresh', methods=['POST'])
+@app.post("/refresh")
 async def refresh():
-    refresh_token = request.json['refresh_token']
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute('SELECT * FROM refresh_tokens WHERE refresh_token = %s', (refresh_token,))
-            token = await cur.fetchone()
-            await cur.close()
-            pool.close()
-    if not token:
-        return jsonify({'error': 'Invalid refresh token.'}), 401
+    refresh_token = request.json["refresh_token"]
     try:
-        jwt.decode(refresh_token, app.config['SECRET_KEY'], algorithms=['HS256'])
+        token_payload = jwt.decode(refresh_token, app.config["SECRET_KEY"], algorithms=["HS256"])
     except jwt.ExpiredSignatureError:
-        return jsonify({'error': 'Refresh token has expired.'}), 401
-    access_token = jwt.encode({'user_id': token[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=60)},
-                              app.config['SECRET_KEY'],
-                              algorithm='HS256')
-    refresh_token = jwt.encode({'user_id': token[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(days=7)},
-                               app.config['SECRET_KEY'], algorithm='HS256')
-    access_expire = datetime.datetime.utcnow() + datetime.timedelta(minutes=60)
-    refresh_expire = datetime.datetime.utcnow() + datetime.timedelta(days=7)
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute('INSERT INTO refresh_tokens (user_id, refresh_token) VALUES (%s, %s)',
-                              (token[0], refresh_token))
-            await cur.close()
-            pool.close()
-    return jsonify({'access_token': access_token, 'refresh_token': refresh_token, 'access_expire': access_expire,
-                    'refresh_expire': refresh_expire}), 200
+        return jsonify({"error": "Refresh token has expired."}), 401
+    except jwt.InvalidTokenError:
+        return jsonify({"error": "Invalid refresh token."}), 401
+    return create_and_return_token(token_payload["user_id"])
+
+
+def create_and_return_token(user_id):
+    access_expire = int(time.time()) + 3600
```

Comment:
Don't use direct operations with time objects. Instead, use the DateTime library.

`access_expire = (datetime.now() + timedelta(hours=1)).timestamp()`